assets/queries/terraform/aws/s3_bucket_policy_accepts_http_requests/query.rego

```diff
@@ -41,11 +41,16 @@ CxPolicy[result] {
 
 validActions := {"*", "s3:*", "s3:GetObject"}
 
-check_action(action) {
-	is_string(action)
-	action == validActions[x]
+check_action(st) {
+	is_string(st.Action)
+	st.Action == validActions[x]
 } else {
-	action[a] == validActions[x]
+	st.Action[a] == validActions[x]
+} else {
+	is_string(st.Actions)
+	st.Actions == validActions[x]
+} else {
+	st.Actions[a] == validActions[x]
 }
```

Comment:
@LupovichRan, sorry for requesting another change. After talking with our security team, we realize that all the s3 actions should be under secure transport. I suggest something like, if you do not mind:

```suggestion
any_s3_action(action) {
	any([action == "*", startswith(action, "s3:")])
}

check_action(st) {
	is_string(st.Action)
	any_s3_action(st.Action)
} else {
	any_s3_action(st.Action[a])
} else {
	is_string(st.Actions)
	any_s3_action(st.Actions)
} else {
	any_s3_action(st.Actions[a])
}
```

Reply:
Back to my first commit suggestions, why do we need this function anyways? Can't we just remove it? Won't it be same as "all actions"?

Reply:
I suggest removing this check, WDYT?

Reply:
I thought about it, but it could return FP if the policy targets another kind of action (which is an edge case, because it will not make sense in the context). WDYT, @LupovichRan?

Reply:
What kind of edge case are you thinking of, as I understand this check will kick off only s3 resource, checking only s3 bucket policy resources, can you provide example of such edge case?

Reply:
Yes, of course, @LupovichRan.

It is an unlikely edge case, but a user could attach a "wrong" policy (with no S3 action) to an S3 bucket. For example:

```
data "aws_iam_policy_document" "neg6" {

  statement {
    effect = "Allow"

    principals {
      type        = "*"
      identifiers = ["*"]
    }

    actions = [
      "iam:*AccessKey*",
    ]


    resources = [
      "arn:aws:s3:::a/*",
      "arn:aws:s3:::a",
    ]
    condition {
      test     = "Bool"
      variable = "aws:SecureTransport"
      values   = ["true"]
    }
  }
}


resource "aws_s3_bucket" "neg6" {
  bucket = "a"
  policy = data.aws_iam_policy_document.neg6.json
}
```